ext/mixed/js/core.js

```diff
@@ -288,6 +293,45 @@ const yomichan = (() => {
             this.trigger('orphaned', {error});
         }
 
+        getTemporaryListenerResult(eventHandler, userCallback, timeout=null) {
+            let resolve;
+            let reject;
+            const listenerPromise = new Promise((_resolve, _reject) => {
+                resolve = _resolve;
```

Comment:
Just move everything inside of the promise executor. `listenerPromise` isn't used anywhere other than the return value, so that should be safer (in case of exceptions, etc.), and it avoids the need to capture resolve and reject.

Reply:
Yeah, that would be cleaner. I was too excited about using the pattern I recently learned for passing resolve and reject to other methods in a class :smile:

Reply:
https://github.com/FooSoft/yomichan/pull/417/commits/5ce02f6d29b663d7d80c30341127b058563a02a9